### ISSUE TITLE
Output returning invokes aren't flowing secretness/dependencies from inputs to the resulting output

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#17474]
Discovered while investigating the root cause of #17459. Looks like both Node.js and Python are affected. We should confirm the other SDKs aren't affected.

Previously:
- **Secrets:** if you passed a secret as an input to an output-returning invoke, the resulting output would also be a secret
- **Dependencies:**  if you passed a resource output as an input to an output-returning invoke, the resulting output would track that dependency

This appears to no longer be the case for the above bullets with the new implementation.

Similar repro as in #17459. Steps:

```
mkdir azureinvoke && cd azureinvoke
pulumi new azure-python
```

Replace `__main__.py` with the following:

```python
import pulumi
from pulumi_azure_native import storage
from pulumi_azure_native import resources

resource_group = resources.ResourceGroup("resource_group")

storage_account = storage.StorageAccount(
    "sa",
    resource_group_name=resource_group.name,
    sku={
        "name": storage.SkuName.STANDARD_LRS,
    },
    kind=storage.Kind.STORAGE_V2,
)

# result = storage.list_storage_account_keys_output(
#     account_name=pulumi.Output.secret(storage_account.name),
#     resource_group_name=resource_group.name)

# pulumi.export("result", result)
```

Run `pulumi up` to create the resources.

Now uncomment the lines at the bottom and run `pulumi preview`.

- Using `pulumi-azure-native` 2.64.2, the exported result is not shown as a secret (regression)
- Using `pulumi-azure-native` 2.63.0, the exported result is shown as a secret (expected)

**Comment by justinvp** [comment]
Fixed by:
- https://github.com/pulumi/pulumi/pull/17460
- https://github.com/pulumi/pulumi/pull/17479
- https://github.com/pulumi/pulumi/pull/17482
- https://github.com/pulumi/pulumi/pull/17501

**Comment by mikhailshilkov** [comment]
This issue has been shipped in [release v3.136.1](https://github.com/pulumi/pulumi/releases/tag/v3.136.1).
